test/sw_mgmt_spec.rb

```diff
@@ -123,36 +123,56 @@
     let(:base_product) do
       instance_double(Y2Packager::Product, name: "dummy", version: "15.0", arch: "x86_64")
     end
+    let(:architecture) { "aarch64" }
     let(:available_products) { [base_product] }
+    let(:self_update_id) { "DummyProductName" }
+    let(:self_update_version) { "DummyProductVersion" }
+    let(:result) { subject.installer_update_base_product(self_update_id, self_update_version) }
 
     before do
       allow(Y2Packager::Product).to receive(:available_base_products).and_return(available_products)
+      allow(Yast::Arch).to receive(:architecture).and_return(architecture)
     end
 
-    it "returns nil if the given self_update_id is empty" do
-      expect(subject.installer_update_base_product("")).to eq(nil)
+    context "when an empty self_update_id is given" do
+      let(:self_update_id) { "" }
+
+      it "returns nil" do
+        expect(result).to eq(nil)
+      end
     end
 
     context "when there is no base product available" do
       let(:available_products) { [] }
 
-      it "returns nil" do
-        allow(Y2Packager::Product).to receive(:available_base_products).and_return([])
-        expect(subject.installer_update_base_product("self_update_id")).to eq(nil)
+      it "returns the product info hash" do
+        expect(result).to include(
+          "name"         => "DummyProductName",
```

Comment:
I think using `self_update_id` here is better to avoid duplication. If you we need to change it for whatever reason we will need to touch only one place.
```suggestion
          "name"         => self_update_id,
```